src/tests/networking_tests.rs

```diff
@@ -1518,22 +1520,33 @@ pub mod net_tests {
             );
         });
 
-        assert_eq!(
-            cage.send_syscall(socketpair.sock1, str2cbuf("test"), 4, 0),
-            4
-        );
+        let cage3 = cage.clone();
+        let thread_2 = interface::helper_thread(move || {
+            assert_eq!(cage3.send_syscall(socketpair.sock1, str2cbuf("test"), 4, 0), 4);
+
+            let mut buf2 = sizecbuf(15);
+            loop {
+                let result = cage3.recv_syscall(socketpair.sock1, buf2.as_mut_ptr(), 15, 0);
+                if result != -libc::EINTR {
+                    break; // if the error was EINTR, retry the syscall
+                }
+            }
+            let str2 = cbuf2str(&buf2);
+            assert_eq!(str2, "Socketpair Test");
+        });
 
-        let mut buf2 = sizecbuf(15);
         loop {
-            let result = cage.recv_syscall(socketpair.sock1, buf2.as_mut_ptr(), 15, 0);
-            if result != -libc::EINTR {
-                break; // if the error was EINTR, retry the syscall
+            if thread_2.is_finished() {
```

Comment:
Why do you check this and have separate if statements for each?  Why not just join both?  Does the order matter for some reason?

Join seems to block (https://doc.rust-lang.org/std/thread/index.html), so I'm not sure why it needs to work like this...

Reply:
Okay, please explain this in the comments in the test...